File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Queue<Integer> result=new LinkedList<>();
        Queue<Integer> queue=new LinkedList<>();
        for(int i = 0;i<progresses.length;i++){
            if((100-progresses[i])%speeds[i]==0){
                queue.offer((100-progresses[i])/speeds[i]);
            }
            else{
                queue.offer((100-progresses[i])/speeds[i]+1);
            }
        }
        
        while(!queue.isEmpty()){
            int cnt=0;
            int cur=queue.poll();
            cnt+=1;
            while((!queue.isEmpty()) && cur>=queue.peek()){
                queue.poll();
                cnt+=1;
            }
            result.offer(cnt);
        }
        int[] answer=new int[result.size()];
        for(int i = 0;i<answer.length;i++){
            answer[i]=result.poll();
        }
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    result=[]
    days_left=deque()
    for i in range(len(speeds)):
        if (100-progresses[i])%speeds[i]==0:
            days_left.append((100-progresses[i])//speeds[i])
        else:
            days_left.append((100-progresses[i])//speeds[i]+1)
    while days_left:
        cnt=0
        current_day=days_left.popleft()
        cnt+=1
        while len(days_left)>0 and current_day>=days_left[0]:
            cnt+=1
            days_left.popleft()
        result.append(cnt)
    return result      